spec/mongoid/traversable_spec.rb

```diff
@@ -770,5 +770,85 @@ class PreLocalSymDiscriminatorChild < PreLocalSymDiscriminatorParent
         end
       end
     end
+
+    context "when setting an invalid discriminator key on the class level", :focus do 
+      context "when duplicate_fields_exception is true" do
+        before do
+          Mongoid.duplicate_fields_exception = true
+
+          class DuplicateDiscriminatorKeyParent
+            include Mongoid::Document
+            field :dkey, type: String
+          end
+
+          class DuplicateDiscriminatorKeyChild < DuplicateDiscriminatorKeyParent
+          end
+        end
+
+        it "raises an error" do
+          expect do 
+            DuplicateDiscriminatorKeyParent.discriminator_key = "dkey"
+          end.to raise_error(Mongoid::Errors::InvalidField)
+        end
+      end
+
+      context "when the field name conflicts with mongoid's internals" do
+        
+        after do 
+          Person.discriminator_key = nil
+        end
+
+        [:_association, :invalid].each do |meth|
+          context "when the field is named #{meth}" do
+  
+            it "raises an error" do
+              expect {
+                Person.discriminator_key = meth
+              }.to raise_error(Mongoid::Errors::InvalidField)
+            end
+          end
+        end
+      end
+    end
+
+    context "when setting an invalid discriminator key on the global level", :focus do 
+      context "when duplicate_fields_exception is true" do
+        before do
+          Mongoid.duplicate_fields_exception = true
+          Mongoid.discriminator_key = "dkey"
+
+          class GlobalDuplicateDiscriminatorKeyParent
+            include Mongoid::Document
+          end
+
+          class GlobalDuplicateDiscriminatorKeyChild < GlobalDuplicateDiscriminatorKeyParent
+          end
+        end
+
+        it "raises an error" do
+          expect do 
+            GlobalDuplicateDiscriminatorKeyParent.field("dkey")
+          end.to raise_error(Mongoid::Errors::InvalidField)
```

Comment:
Why is this an error?

Reply:
Because on creating a child class that inherits from the parent, a field is created called `dkey` because it is the discriminator key. Then when creating another field called `dkey` it triggers the duplicate field error.

Reply:
:+1: